scipy/spatial/distance.py

```diff
@@ -1994,8 +1996,8 @@ def cdist(XA, XB, metric='euclidean', p=2, V=None, VI=None, w=None):
     XB = np.asarray(XB, order='c')
 
     # The C code doesn't do striding.
-    XA = _copy_array_if_base_present(_convert_to_double(XA))
-    XB = _copy_array_if_base_present(_convert_to_double(XB))
```

Comment:
Looks like there was duplicate conversion - all calls to ``_distance_wrap.cdist_*`` still have the correct conversions further down in this function.